compass/landice/tests/thwaites/mesh.py

```diff
@@ -0,0 +1,218 @@
+                            target='antarctica_1km_2020_10_20_ASE.nc',
+                            database='')
+
+    # no setup() method is needed
+
+    def run(self):
+        """
+        Run this step of the test case
+        """
+        logger = self.logger
+        config = self.config
+        section = config['high_res_mesh']
+
+        logger.info('calling build_cell_width')
+        cell_width, x1, y1, geom_points, geom_edges = self.build_cell_width()
+        logger.info('calling build_planar_mesh')
+        build_planar_mesh(cell_width, x1, y1, geom_points,
+                          geom_edges, logger=logger)
+        dsMesh = xarray.open_dataset('base_mesh.nc')
+        logger.info('culling mesh')
+        dsMesh = cull(dsMesh, logger=logger)
+        logger.info('converting to MPAS mesh')
+        dsMesh = convert(dsMesh, logger=logger)
+        logger.info('writing grid_converted.nc')
+        write_netcdf(dsMesh, 'grid_converted.nc')
+
+        levels = section.get('levels')
+
+        logger.info('calling create_landice_grid_from_generic_MPAS_grid.py')
+        args = ['create_landice_grid_from_generic_MPAS_grid.py',
+                '-i', 'grid_converted.nc',
+                '-o', 'ase_1km_preCull.nc',
+                '-l', levels, '-v', 'glimmer']
+        check_call(args, logger=logger)
+
+        # This step uses a subset of the whole Antarctica dataset trimmed to
+        # the Amundsen Sean Embayment, to speed up interpolation.
+        # This could also be replaced with the full Antarctic Ice Sheet
+        # dataset.
+        logger.info('calling interpolate_to_mpasli_grid.py')
+        args = ['interpolate_to_mpasli_grid.py', '-s',
+                'antarctica_1km_2020_10_20_ASE.nc', '-d',
+                'ase_1km_preCull.nc', '-m', 'b', '-t']
+
+        check_call(args, logger=logger)
+
+        # This step is only necessary if you wish to cull a certain
+        # distance from the ice margin, within the bounds defined by
+        # the GeoJSON file.
+        cullDistance = section.get('cull_distance')
+        if float(cullDistance) > 0.:
+            logger.info('calling define_cullMask.py')
+            args = ['define_cullMask.py', '-f',
+                    'ase_1km_preCull.nc', '-m'
+                    'distance', '-d', cullDistance]
+
+            check_call(args, logger=logger)
+        else:
+            logger.info('cullDistance <= 0 in config file. '
+                        'Will not cull by distance to margin. \n')
+
+        # This step is only necessary because the GeoJSON region
+        # is defined by lat-lon.
+        logger.info('calling set_lat_lon_fields_in_planar_grid.py')
+        args = ['set_lat_lon_fields_in_planar_grid.py', '-f',
+                'ase_1km_preCull.nc', '-p', 'ais-bedmap2']
+
+        check_call(args, logger=logger)
+
+        logger.info('calling MpasMaskCreator.x')
+        args = ['MpasMaskCreator.x', 'ase_1km_preCull.nc',
```

Comment:
@xylar , should this be replaced with the `mask` method in MPAS-Tools' mesh conversion package?

Reply:
I don't think it matters too much.  `mask()` is just a wrapper that takes `xarray.Dataset` and `geometric_features.FeatureCollection` objects as its inputs.  In this case where the files are already written, the wrapper doesn't save work.

But you should maybe consider using the alternative tool: http://mpas-dev.github.io/MPAS-Tools/stable/mesh_conversion.html#mask-creation-with-python-multiprocessing
The tool is called `compute_mpas_region_masks` and seems to work better if called as a subprocess.  It should be significantly faster than `MpasMaskCreator.x` on HPC nodes because it uses multiprocessing.

Reply:
@trhille, do you want to try `compute_mpas_region_masks` or are you pretty convinced that's not a time-limiting step here?

Reply:
@xylar , I added a timer on the call to MpasMaskCreator.x on a 2-8km test, and it only took 9 seconds out of 3:06 total runtime, so I'm happy to leave it as-is for now.